p2p/client.go

```diff
@@ -382,13 +385,14 @@ func (c *Client) setupGossiping(ctx context.Context) error {
 		return err
 	}
 
-	c.txGossiper, err = NewGossiper(c.host, c.ps, c.getTxTopic(), c.logger, WithValidator(c.txValidator))
+	logger := tmlog.NewTMLogger(os.Stdout)
```

Comment:
instead of `tmlog.NewTMLogger` does `logger.With("module", "tx-gossip")` works?

Reply:
added the change but it's not exactly the same. Which logger object would you use for `With`?

Reply:
just use `c.logger.With`?

Reply:
That would reintroduce the problem. Not using `c.logger` is the fix for the issue: https://github.com/rollkit/rollkit/issues/910#issuecomment-1535788972